test/unit/mdc-tabs/mdc-tab-bar-foundation.test.js

```diff
@@ -0,0 +1,140 @@
+/**
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and * limitations under the License.
+ */
+
+import {assert} from 'chai';
+import td from 'testdouble';
+
+import {setupFoundationTest} from '../helpers/setup';
+import {verifyDefaultAdapter} from '../helpers/foundation';
+import {createMockRaf} from '../helpers/raf';
+import {supportsCssVariables} from '../../../packages/mdc-ripple/util';
+
+import MDCTabBarFoundation from '../../../packages/mdc-tabs/tab-bar/foundation';
+
+suite('MDCTabBarFoundation');
+
+test('exports cssClasses', () => {
+  assert.isOk('cssClasses' in MDCTabBarFoundation);
+});
+
+test('default adapter returns a complete adapter implementation', () => {
+  verifyDefaultAdapter(MDCTabBarFoundation, [
+    'addClass', 'removeClass', 'bindOnMDCTabSelectedEvent',
+    'unbindOnMDCTabSelectedEvent', 'registerResizeHandler',
+    'deregisterResizeHandler', 'getOffsetWidth', 'setStyleForIndicator',
+    'getOffsetWidthForIndicator', 'notifyChange',
+    'getNumberOfTabs', 'isTabActiveAtIndex', 'setTabActiveAtIndex',
+    'isDefaultPreventedOnClickForTabAtIndex',
+    'setPreventDefaultOnClickForTabAtIndex', 'measureTabAtIndex',
+    'getComputedWidthForTabAtIndex', 'getComputedLeftForTabAtIndex',
+  ]);
+});
+
+function setupTest() {
+  const {foundation, mockAdapter} = setupFoundationTest(MDCTabBarFoundation);
+  const {UPGRADED} = MDCTabBarFoundation.cssClasses;
+  const {TAB_SELECTOR, INDICATOR_SELECTOR} = MDCTabBarFoundation.strings;
+
+  return {foundation, mockAdapter, UPGRADED, TAB_SELECTOR, INDICATOR_SELECTOR};
+}
+
+test('#init adds upgraded class to tabs', () => {
+  const {foundation, mockAdapter, UPGRADED} = setupTest();
+
+  foundation.init();
+  td.verify(mockAdapter.addClass(UPGRADED));
+});
+
+test('#init registers listeners', () => {
+  const {foundation, mockAdapter} = setupTest();
+  const {isA} = td.matchers;
+
+  foundation.init();
+  td.verify(mockAdapter.bindOnMDCTabSelectedEvent());
+  td.verify(mockAdapter.registerResizeHandler(isA(Function)));
+});
+
+if (supportsCssVariables(window)) {
```

Comment:
I don't think this needs to be wrapped in this if statement because this logic doesn't use custom variables...?